vcd/resource_vcd_rights_bundle.go

```diff
@@ -0,0 +1,313 @@
+	})
+	if err != nil {
+		return diag.Errorf("[rights bundle create] error creating role %s: %s", rightsBundleName, err)
+	}
+	if len(inputRights) > 0 {
+		err = rightsBundle.AddRights(inputRights)
+		if err != nil {
+			return diag.Errorf("[rights bundle create] error adding rights to rights bundle %s: %s", rightsBundleName, err)
+		}
+	}
+
+	inputTenants, err := getTenants(vcdClient, "rights bundle create", d)
+	if err != nil {
+		return diag.Errorf("%s", err)
+	}
+	if publishToAllTenants {
+		err = rightsBundle.PublishAllTenants()
+		if err != nil {
+			return diag.Errorf("[rights bundle create] error publishing to all tenants - rights bundle %s: %s", rightsBundleName, err)
+		}
+	}
+	if len(inputTenants) > 0 {
+		err = rightsBundle.PublishTenants(inputTenants)
+		if err != nil {
+			return diag.Errorf("[rights bundle create] error publishing to tenants - rights bundle %s: %s", rightsBundleName, err)
+		}
+	}
+	return genericRightsBundleRead(ctx, d, meta, "resource", "create")
+}
+
+func resourceRightsBundleRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	return genericRightsBundleRead(ctx, d, meta, "resource", "read")
+}
+
+func genericRightsBundleRead(ctx context.Context, d *schema.ResourceData, meta interface{}, origin, operation string) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+
+	rightsBundleName := d.Get("name").(string)
+
+	rightsBundle, err := vcdClient.Client.GetRightsBundleByName(rightsBundleName)
+	if err != nil {
+		if origin == "resource" {
+			d.SetId("")
+			return nil
+		}
+		return diag.Errorf("[rights bundle read-%s] error retrieving rights bundle %s: %s", operation, rightsBundleName, err)
+	}
+
+	d.SetId(rightsBundle.RightsBundle.Id)
+	_ = d.Set("description", rightsBundle.RightsBundle.Description)
+	_ = d.Set("bundle_key", rightsBundle.RightsBundle.BundleKey)
+	_ = d.Set("read_only", rightsBundle.RightsBundle.ReadOnly)
+
+	rights, err := rightsBundle.GetRights(nil)
+	if err != nil {
+		return diag.Errorf("[rights bundle read-%s] error while querying rights bundle rights: %s", operation, err)
+	}
+	var assignedRights []interface{}
+
+	for _, right := range rights {
+		assignedRights = append(assignedRights, right.Name)
+	}
+	err = d.Set("rights", assignedRights)
+	if err != nil {
+		return diag.Errorf("[rights bundle read-%s] error setting rights for rights bundle %s: %s", operation, rightsBundleName, err)
+	}
+
+	tenants, err := rightsBundle.GetTenants(nil)
+	if err != nil {
+		return diag.Errorf("[rights bundle read-%s] error while querying rights bundle tenants: %s", operation, err)
+	}
+	var registeredTenants []interface{}
+
+	publishAll := false
+	if rightsBundle.RightsBundle.PublishAll != nil {
+		publishAll = *rightsBundle.RightsBundle.PublishAll
+	}
+	_ = d.Set("publish_to_all_tenants", publishAll)
+	for _, tenant := range tenants {
+		registeredTenants = append(registeredTenants, tenant.Name)
+	}
+	if !*rightsBundle.RightsBundle.PublishAll {
```

Comment:
Should this be such to again avoid nil pointer de-reference risk (just as above)
```suggestion
	if !publishAll {
```